### ISSUE TITLE
False negatives in unused exports

### ISSUE THREAD

**Issue by neelmraman** [webpro-nl/knip#695]
Hi, thanks for your work on this project. I am using the latest version of knip (5.22.3) and found some false negatives where unused exports were not reported as unused.

**Issue 1: name is a substring of a different name when using `ignoreExportsUsedInFile`**
```typescript
// file: a.ts
export const unusedFunction = () : string => {
    return 'a';
};

export const unusedFunctionUnrelated = () : string => {
    return 'b';
};

unusedFunctionUnrelated()
```
Knip correctly reports that both functions above are unused when `ignoreExportsUsedInFile` is false, but does not report either function when it is set to true. If I rename `unusedFunctionUnrelated` to something else that does not contain `unusedFunction` within it, knip then correctly reports `unusedFunction` as unused and the other function as used since it is called within the module.

**Issue 2: multiline named exports when using `ignoreExportsUsedInFile`**
```typescript
const abc = '123'
const def = '456'
export {
  abc, 
  def
}
```
Knip correctly reports that both constants above are unused when `ignoreExportsUsedInFile` is false, but does not report either constant when it is set to true. Interestingly, knip does report both constants with `ignoreExportsUsedInFile` set to true as long I remove the multiline exports via
```typescript
const abc = '123'
const def = '456'
export { abc, def }
```
or 
```typescript
const abc = '123'
const def = '456'
export { abc }
export { def }
```

**Issue 3: Re-exporting an unused export**
If I create a new file b.ts which contains
```typescript
export { unusedFunction } from './a'
```
and then my `index.ts` is
```typescript
export * from './a'
export * from './b'
```
`unusedFunction` is not reported as unused. If I remove `export * from './b'` from my `index.ts`, making it so that b.ts is unreachable, knip will then report `unusedFunction` as unused. None of the flags (ignoreExportsUsedInFile and includeEntryExports) seem to affect this behavior.

**Comment by webpro** [comment]
In https://knip.dev/guides/issue-reproduction:

- Please create minimal issue reproductions
- Please create separate issues

Let's keep this one for only the first issue you've mentioned. No need to create a repro for this first one, I'm fixing this one now.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.23.0. See [Release 5.23.0](https://github.com/webpro-nl/knip/releases/tag/5.23.0) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
